src/odin/worker.cc

```diff
@@ -66,18 +78,30 @@ odin_worker_t::work(const std::list<zmq::message_t>& job,
                                  boost::optional<std::string>("Failed parsing pbf in Odin::Worker")};
     }
 
-    // narrate them and serialize them along
-    narrate(request);
-    auto response = tyr::serializeDirections(request);
-    const bool as_gpx = request.options().format() == Options::gpx;
-    return to_response(response, info, request, as_gpx ? worker::GPX_MIME : worker::JSON_MIME,
-                       as_gpx);
+    // its either a simple status request or its a route to narrate
+    switch (request.options().action()) {
+      case Options::status: {
+        return to_response(status(request), info, request);
+      }
+      default: {
+        // narrate them and serialize them along
```

Comment:
this code cov is nuts! literally all the tests would fail if it didnt get in here

Reply:
oh actually this is the service and not the actor so scratch that this code cov is correct :smile: its because we dont actually run the service against actual data in unit tests. we totally could and should though against utrecht. ill add an issue for that and we can coerce the `loki_service` test into more of an integration test